cmd/main_test.go

```diff
@@ -476,3 +478,116 @@ var _ = DescribeTable("GetWorkspacesWithAccess querying for workspaces with acce
 		})
 	})
 })
+
+var _ = Describe("GetUserNamespaces", func() {
+	var e *echo.Echo
+	var req *labels.Requirement
+	var expectedNamespaces []string
+
+	BeforeEach(func() {
+		e = echo.New()
+	})
+
+	Context("When querying for a specific user namespace", func() {
+		BeforeEach(func() {
+			var err error
+			req, err = labels.NewRequirement("kubernetes.io/metadata.name", selection.In, []string{"test-tenant"})
+			Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+			expectedNamespaces = []string{"test-tenant"}
+		})
+
+		It("Should return only the specified namespace", func() {
+			namespaces, err := getUserNamespaces(e, *req)
+			Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+			var actualNamespaces []string
+			for _, ns := range namespaces {
+				actualNamespaces = append(actualNamespaces, ns.Name)
+			}
+
+			Expect(actualNamespaces).To(ConsistOf(expectedNamespaces))
+		})
+
+		AfterEach(func() {
+			req = nil
+			expectedNamespaces = nil
+		})
+	})
+
+	Context("When querying for multiple specific user namespaces", func() {
+		BeforeEach(func() {
+			var err error
+			req, err = labels.NewRequirement("kubernetes.io/metadata.name", selection.In, []string{"test-tenant", "test-tenant-2"})
+			Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+			expectedNamespaces = []string{"test-tenant", "test-tenant-2"}
+		})
+
+		It("Should return both specified namespaces", func() {
+			namespaces, err := getUserNamespaces(e, *req)
+			Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+			var actualNamespaces []string
+			for _, ns := range namespaces {
+				actualNamespaces = append(actualNamespaces, ns.Name)
+			}
+
+			Expect(actualNamespaces).To(ConsistOf(expectedNamespaces))
+		})
+
+		AfterEach(func() {
+			req = nil
+			expectedNamespaces = nil
+		})
+	})
+
+	Context("When querying for a non-existent namespace", func() {
+		BeforeEach(func() {
+			var err error
+			req, err = labels.NewRequirement("kubernetes.io/metadata.name", selection.In, []string{"non-existent-namespace"})
+			Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+			expectedNamespaces = []string{}
+		})
+
+		It("Should return an empty list", func() {
+			namespaces, err := getUserNamespaces(e, *req)
+			Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+			Expect(namespaces).To(BeEmpty())
+		})
+
+		AfterEach(func() {
+			req = nil
+			expectedNamespaces = nil
+		})
+	})
+
+	// This test case alone passes becasue getUserNamespaces function is always returning all user namespaces regardless of the input requirement.
```

Comment:
It does not. `getUserNamespaces` return all namespaces with the `konflux.ci/type`(=user) label and any additional labels you've given it. 
if it returns all namespaces, something in the resource management for the test is wrong.

Reply:
Okay will check that out